### ISSUE TITLE
consider remote galleys and pub ids in native import and export

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#1713]

**Comment by bozana** [comment]
@asmecher, what exactly mean the "advice" attribute? -- At the moment all id elements have advice=ignore. The internal ids are ignored, but the publisher-id and doi/urn are stored (although advice=ignore). Is this behavior correct?
For some reason the doi/urn are not stored, because the lib/pkp/classes/plugins/PKPPubIdPlugin.inc.php registration for the hook getAdditionalFieldNames is not called :-( Any idea why? It is called and everything is fine when assigning the pub ids in the workflow, but here, when importing the objects, it is not :-( 
THANKS!!!

**Comment by asmecher** [comment]
Maybe the pub ID plugin category isn't loaded during import?

The `advice` attribute is something I'd like to build on later to allow importers to instruct OJS what to do with specified IDs: update existing entities that have that ID, or just ignore it? For now it's hard-coded to `ignore`, as you've noted, because that support hasn't been added yet.

**Comment by bozana** [comment]
Of course... me crazy... :-\ Thanks!!!

**Comment by bozana** [comment]
@asmecher, what if the user (or maybe also the user group) from the import XML file doesn't exist in the DB? At the moment NULL is inserted. Is NULL allowed? If so, then the assertions on other places in the system shouldn't be there, e.g. here: https://github.com/pkp/pkp-lib/blob/master/plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php#L110. But actually I believe it is better to insert a user and a user group. Maybe the one from the journal manager inserting the file? Or to return the error message? Hmmm...

**Comment by bozana** [comment]
pkp-lib PR: https://github.com/pkp/pkp-lib/pull/1719
ojs PR: https://github.com/pkp/ojs/pull/970
@asmecher, could you please review those PRs? The tests will go through once this PR is merged: https://github.com/pkp/ojs/pull/965. Thus, then I will rebase this...

**Comment by asmecher** [comment]
Merged and cherry-picked as needed, thanks! This needs implementation for OMP, right? If so, feel free to change the milestone to OMP 1.2.1 and leave the issue open.

**Comment by bozana** [comment]
@asmecher, thanks a lot! No, I think this does not need any implementation for OMP, because all objects (Submission, Representation and Files) are considered in pkp-lib. OJS needed only changes for issue. I will double check it, but I think I've already tested it... THANKS!!!

**Comment by asmecher** [comment]
OK, thanks! Close whenever you're ready.

**Comment by asmecher** [comment]
From https://github.com/pkp/pkp-lib/issues/1713#issuecomment-239407588:
> @asmecher, what if the user (or maybe also the user group) from the submission_file revision element from the import XML file doesn't exist in the DB?

Hmm, `null` should definitely not be allowed -- we need to know who is responsible for the file. Do we have enough information to create a user? I suspect probably not. I would suggest, for the moment, failing if the user does not exist; longer term, we may need to merge some capabilities from the user import/export plugin in, permitting references in the XML from files to user entries that more fully describe the user.

**Comment by bozana** [comment]
@asmecher, I moved the comment above in a new issue (https://github.com/pkp/pkp-lib/issues/1728). The OMP does not need changes, thus closing...

**Comment by asmecher** [comment]
OK, thanks!
